### PR TITLE
added pundit gem and created the user_policy

### PR DESCRIPTION
**Pull request by alanlamaswolox**
## Summary
Added pundit gem 

## Trello
https://trello.com/c/vxaJNYf1/12-21-instalar-y-configurar-pundit

### CODE REVIEW

File: app/policies/application_policy.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+class ApplicationPolicy
```
Comment (nicozar95) [inline app/policies/application_policy.rb]: Do u need this?

File: app/policies/user_policy.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class UserPolicy < ApplicationPolicy
+  attr_reader :user, :passed_user
+
+  def initialize(user, passed_user)
+    @user = user
+    @passed_user = passed_user
+  end
+
```
Comment (nicozar95) [inline app/policies/user_policy.rb:8]: Don't you need to add a a create? authorization

File: app/controllers/api/v1/rents_controller.rb [hunk 3]
```diff
@@ -3,7 +3,13 @@ module V1
...
+        @user = User.find(params[:user_id])
+        begin
+          authorize(@user)
+          render_paginated @rents
+        rescue Pundit::NotAuthorizedError
+          render json: { errors: "You're not authorized" }, status: :unauthorized
+        end
       end
 
       def create
```
Comment (nicozar95) [inline app/controllers/api/v1/rents_controller.rb:13]: What about the `create` ???? a User "A" can create rents for a User "B" ?

**Review by nicozar95 — CHANGES_REQUESTED** [review]

File: app/policies/application_policy.rb (hunk 1 above)
Comment (alanlamaswolox) [reply app/policies/application_policy.rb]: The documentation said that the all policy's classes should inherit from that class, but i don't know exactly

File: .rubocop.yml [hunk 4]
```diff
@@ -8,3 +8,7 @@ Metrics/BlockLength:
   Exclude:
     - 'spec/**/*.rb'
     - 'app/admin/*.rb'
+
+Metrics/MethodLength:
```
Comment (nicozar95) [inline .rubocop.yml]: Dont' do this, dont exclude a method length in all of the controllers. Which method gives you problem?

File: app/controllers/api/v1/rents_controller.rb [hunk 5]
```diff
@@ -3,15 +3,27 @@ module V1
     class RentsController < ApiController
       def index
         @rents = Rent.where(user_id: params[:user_id])
-        render_paginated @rents
+        @user = User.find(params[:user_id])
+        begin
+          authorize(@user)
+          render_paginated @rents
+        rescue Pundit::NotAuthorizedError
```
Comment (nicozar95) [inline app/controllers/api/v1/rents_controller.rb]: It's not necesary to rescue the error, pundit sent it by default.

File: app/controllers/api/v1/rents_controller.rb [hunk 6]
```diff
@@ -3,15 +3,27 @@ module V1
...
-          render json: { errors: @rent.errors.full_messages }, status: :bad_request
+        @user = User.find(params[:user_id])
+        begin
+          authorize(@user)
+          if @rent.save
+            EmailWorker.perform_async(@rent.id)
+          else
+            render json: { errors: @rent.errors.full_messages }, status: :bad_request
+          end
+        rescue Pundit::NotAuthorizedError
```
Comment (nicozar95) [inline app/controllers/api/v1/rents_controller.rb]: Same as above

**Review by nicozar95 — CHANGES_REQUESTED** [review]
Change pls!

File: .rubocop.yml (hunk 4 above)
Comment (alanlamaswolox) [reply .rubocop.yml]: The create method of rents_controller.rb. What should i do with rubocop's concerns about the method's length?

Comment (nicozar95) [reply .rubocop.yml]: Try to fix it so you don't need to disable the method length in ALL THE CONTROLLERS, maybe adding some new spaces for example, if you have troubles with this let me know

File: app/controllers/api/v1/rents_controller.rb [hunk 7]
```diff
@@ -20,6 +26,10 @@ def create
       def rent_params
         params.require(:rent).permit(:user_id, :book_id, :initial_date, :final_date)
       end
+
+      def pundit_not_authorized_error
```
Comment (nicozar95) [inline app/controllers/api/v1/rents_controller.rb:30]: Make this and the `rent_params`methods private

**Review by nicozar95 — CHANGES_REQUESTED** [review]

**Review by nicozar95 — APPROVED** [review]
Sorry!

**Pull request merged**